common/src/warp_runner/manager/commands/multipass_commands.rs

```diff
@@ -395,6 +400,23 @@ pub async fn handle_multipass_cmd(cmd: MultiPassCmd, warp: &mut super::super::Wa
                     rsp.send(Err(e))
                 }
             };
+        }
+        MultiPassCmd::GetIdentity { did, rsp } => {
+            let pic = warp.multipass.identity_picture(&did).await;
+            let r = warp.multipass.get_identity(Identifier::DID(did)).await; //(dids, &mut warp.multipass).await;
```

Comment:
Maybe use `map` and `and_then` chain to have something to supply to `rsp`?

Reply:
dont really see a point specially since gotta check if returned ids are empty or not

Reply:
Not really as you could do something like

```rust
let id = warp.multipass.get_identity(Identifier::DID(did))
        .await
        .map(|list| list.first().cloned())
        .and_then(|id| id.ok_or(Error::IdentityDoesntExist));
```

Especially since youre not expecting a list but a single identity.

Reply:
hmm you cant use await in a closure so calling e.g. `warp.multipass.get_identity(Identifier::DID(did))` in a mapping operation doesn't really work? so you still need to do something like `match`

Reply:
Although it is possible to await in a map, In this case you wouldnt be since `MultiPass::get_identity` returns `Vec<Identity>` and not a future, unless im misunderstanding your question.

Reply:
oh sry i meant like `identity_picture`. (though both return a future)

Reply:
is there a benefit for `map` and `and_then` or is it just for readability/looking nice?

Reply:
Just readability and reducing the amount of indents.